models/amazon_selling_partner__order_items.sql

```diff
@@ -0,0 +1,116 @@
+{{ config(enabled=var('amazon_selling_partner__using_orders_module', true)) }}
+
+with orders as (
+
+    select *
+    from {{ ref('stg_amazon_selling_partner__orders') }}
+),
+
+order_item as (
+
+    select *
+    from {{ ref('stg_amazon_selling_partner__order_item') }}
+),
+
+{% if var('amazon_selling_partner__using_catalog_module', true) %}
+{# Open Q: Should we bring item in here?  #}
+item as (
+
+    select *
+    from {{ ref('int_amazon_selling_partner__item') }}
+),
+{% endif %}
+
+order_item_promotion_id as (
+    
+    select *
+    from {{ ref('stg_amazon_selling_partner__order_item_promotion_id') }}
+),
+
+aggregate_promotions as (
+
+    select 
+        source_relation,
+        amazon_order_id,
+        order_item_id,
+        count(distinct promotion_id) as count_promotions_used
+
+    from order_item_promotion_id 
+    group by 1,2,3
+),
+
+joined as (
+
+    select 
+        order_item.*,
+        {# Open Q: What other order/header level fields would be useful to have here? #}
```

Comment:
This is a good place to start for now. We can always add more upon request

Reply:
sounds good, removed comment